tests/printer/other/ocamlString.res

```diff
@@ -0,0 +1,27 @@
+let x = "\132\149\166"
+
+let s = "\123 \o111 \xA0"
+
+let x = "foo\010bar"
+let x = "foo\x0Abar"
+let x = "foo\o012bar"
+
+let x = "😁 this works now 😆"
+let x = `😁 this works now 😆`
+
+
+// The `//` should not result into an extra comment
+let x = j`https://www.apple.com`
+let x = `https://www.apple.com`
+let x = js`https://www.apple.com`
+let x = "https://www.apple.com"
+let x = sql`https://www.apple.com`
```

Comment:
The reason I ended up converting this file from .ml to .res is that when parsing the .ml file this is parsed as a string (with prefix `sql`) where as when it's a .res file we parse it as an apply node with the callExpr being `sql`.

While we could have the `res` parser work the same as the `ml` parser and then do an AST transform to cover the string to a function call, it would be a pretty complicate transform to do since template strings end up getting parsed as nested `^` calls if there are any interpolations.

Also, if ReScript is supposed to be independent of OCaml, I'm not sure what the benefit is of trying to keep the parse trees we generate for the two in sync.